src/qdrant_client/version_check.rs

```diff
@@ -0,0 +1,128 @@
+use std::error::Error;
+use std::fmt;
+
+use semver::Version;
+
+pub fn parse(version: &str) -> Result<Version, VersionParseError> {
+    if version.is_empty() {
+        return Err(VersionParseError::EmptyVersion);
+    }
+    match Version::parse(version) {
+        Ok(v) => Ok(v),
+        Err(_) => Err(VersionParseError::InvalidFormat(version.to_string())),
+    }
+}
+
+#[derive(Debug)]
+pub enum VersionParseError {
+    EmptyVersion,
+    InvalidFormat(String),
+}
+
+impl fmt::Display for VersionParseError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            VersionParseError::EmptyVersion => write!(f, "Version is empty"),
+            VersionParseError::InvalidFormat(version) => {
+                write!(
+                    f,
+                    "Unable to parse version, expected format: x.y[.z], found: {}",
```

Comment:
No strong feeling about this, pick whichever you like best:

```suggestion
                    "Unable to parse version, expected semver format, found: {}",
```

Reply:
Let's keep the `format: x.y[.z]` as it is the same in all other clients.